.github/workflows/authors.yml

```diff
@@ -0,0 +1,38 @@
+name: New Author Check
+
+on:
+  - pull_request
+
+jobs:
+  check-id:
+    name: New Author Check
+    runs-on: ubuntu-latest
+    steps:
+      - run: sudo apt -y install git
+      - run: git clone https://github.com/${{ github.repository }}
+      - run: git remote add author ${{ github.event.pull_request.head.repo.html_url }}
+        working-directory: suricata
+      - run: git fetch author
+        working-directory: suricata
+      - run: git checkout author/${{ github.event.pull_request.head.ref }}
+        working-directory: suricata
+      - name: Export known authors from master branch
+        run: git log --format="%an <%ae>" origin/master | sort | uniq > ../authors.txt
+        working-directory: suricata
+      - name: Export authors from new commits
+        run: git log --format="%an <%ae>" origin/${GITHUB_BASE_REF}... | sort | uniq > ../commit-authors.txt
+        working-directory: suricata
+      - name: Check new authors
+        run: |
+          touch new-authors.txt
+          while read -r author; do
+             echo "Checking author: ${author}"
+             if ! grep -q "^${author}\$" authors.txt; then
+                 echo "ERROR: ${author} NOT FOUND"
+                 echo "::error ::New author found: ${author}"
```

Comment:
what happens if you use `::warning` here and then not `exit 1`?

Reply:
Running 2 tests with exit 1 removed:
- https://github.com/OISF/suricata/pull/8848 (::error)
- https://github.com/OISF/suricata/pull/8849 (::warning)

I believe the exit is the only way to see any notification on the main pull request page, without going to the summary and looking for annotations.

Reply:
yeah I see, thanks